packages/zoom-in-ui/src/components/app.js

```diff
@@ -0,0 +1,230 @@
+import 'url-search-params-polyfill'
+import AddImageForm from './form'
+import EmbeddedCode from './embedded-code'
+import PropTypes from 'prop-types'
+import React, { useState } from 'react'
+import Setting from './setting'
+import themes from '../themes'
+import webpackAssets from '@twreporter/zoom-in/dist/webpack-assets.json'
+import zoomIn from '@twreporter/zoom-in'
+// lodash
+import get from 'lodash/get'
+import map from 'lodash/map'
+import merge from 'lodash/merge'
+// @material-ui
+import ArrowDownwardIcon from '@material-ui/icons/ArrowDownward'
+import Button from '@material-ui/core/Button'
+import Container from '@material-ui/core/Container'
+import CssBaseline from '@material-ui/core/CssBaseline'
+import DevicesIcon from '@material-ui/icons/Devices'
+import Divider from '@material-ui/core/Divider'
+import Paper from '@material-ui/core/Paper'
+import RedoIcon from '@material-ui/icons/Redo'
+import Typography from '@material-ui/core/Typography'
+import { makeStyles } from '@material-ui/core/styles'
+
+const _ = {
+  get,
+  map,
+  merge,
+}
+
+const searchKey = {
+  image: 'image',
+  caption: 'caption',
+}
+
+const useStyles = makeStyles(theme => ({
+  container: {
+    textAlign: 'right',
+    padding: 0,
+  },
+  paper: {
+    padding: '50px',
+    marginTop: '50px',
+  },
+  divider: {
+    margin: '20px 0',
+  },
+  sandbox: {
+    width: '100%',
+    marginTop: '25px',
+  },
+  button: {
+    margin: `${theme.spacing(2)}px 0`,
+  },
+  alignRight: {
+    textAlign: 'right',
+  },
+}))
+
+const getParamsFromSearch = query => {
+  if (typeof window !== 'undefined') {
+    const searchParams = new URLSearchParams(window.location.search)
+    return searchParams.get(query)
+  }
+}
+
+const adoptTwreporterTheme = (image, caption) => image && caption
+
+const App = props => {
+  const { description, title } = props
+  const classes = useStyles()
+  const [imageLink, setImageLink] = useState(
+    getParamsFromSearch(searchKey.image) || ''
+  )
+  const [imageCaption, setImageCaption] = useState(
+    getParamsFromSearch(searchKey.caption) || ''
+  )
+  const [code, setCode] = useState(null)
+  const [buildCodeError, setBuildCodeError] = useState(null)
+  const [theme, setTheme] = useState(
+    _.merge(
+      {},
+      adoptTwreporterTheme(imageLink, imageCaption)
+        ? themes.twreporterTheme
+        : themes.defaultTheme
+    )
+  )
+
+  const handleSaveLink = ({ imgUrl, caption }) => {
+    const trimmedLink = imgUrl.trim()
+    if (trimmedLink.length > 0) {
+      const newTheme = adoptTwreporterTheme(trimmedLink, caption)
+        ? themes.twreporterTheme
+        : themes.defaultTheme
+      setImageLink(trimmedLink)
+      setImageCaption(caption)
+      setTheme(_.merge({}, newTheme))
```

Comment:
Does `_.merge({}, newTheme)` aim to create a new object with same data?

Reply:
Yes, but this seems redundant, I'll remove it then.